File: 프로그래머스/2/42860. 조이스틱/조이스틱.java
class Solution {
    public int solution(String name) {
        int leftRight = name.length() - 1;
        int upDown = 0;

        for (int i = 0; i < name.length(); i++) {
            int idx = i + 1;

            while (idx < name.length() && name.charAt(idx) == 'A') {
                idx++;
            }

            leftRight = Math.min(leftRight, Math.min(i * 2 + name.length() - idx, i + (name.length() - idx) * 2));
            upDown += Math.min(name.charAt(i) - 'A', 'Z' + 1 - name.charAt(i));
        }

        return leftRight + upDown;
    }
}

File: 프로그래머스/lv2/42860. 조이스틱/조이스틱.kt
class Solution {
    fun solution(name: String): Int {
        var leftRight = name.length - 1
        var upDown = 0

        name.forEachIndexed { idx, c ->
            var idxA = idx + 1

            while (idxA < name.length && name[idxA] == 'A') {
                idxA++
            }

            leftRight = minOf(leftRight, 2 * idx + name.length - idxA, idx + 2 * (name.length - idxA))
            upDown += minOf(c - 'A', 'Z' - c + 1)
        }

        return leftRight + upDown
    }
}